src/transformers/trainer.py

```diff
@@ -240,6 +244,22 @@ def __init__(
         if model is None and model_init is not None:
             model = self.call_model_init()
 
+        if args.deepspeed:
+            model_parameters = filter(lambda p: p.requires_grad, model.parameters())
+            model, optimizer, training_dataloader, lr_scheduler = deepspeed.initialize(
+                args=args,
```

Comment:
It can handle all our `TrainingArguments` out of the box?

Reply:
I think it just ignores anything it doesn't need. So far I actually have only encountered only 2 `--deepspeed --deepspeed_config ds_config.json` - the rest comes from their config file. 

I wonder why they even needed 2 args, I'd just use `--deepspeed ds_config.json` as it'd be always True and have the config file in it ;)

Reply:
If it only uses two, I'd say we create a wrapper that only contains those two. By reading the current code, I expect `deepspeed` to consume and make use of all my training arguments, so it deserves at least a comment.
Creating our own wrapper would help with the naming of the args (see previous comment).

Reply:
I think this is going to change at least a bit. Please see: https://github.com/microsoft/DeepSpeed/issues/610#issuecomment-749621750

We need to have one place to define the batch size. Currently we have to do it twice (cl arg + ds_config.js) - because we handle the datasets and ds handles the rest where bs size is required. We could delegate dataset handling to DS completely but it'll make our code more complicated.

So we need to find a way to pass our cl arg of batch size to deepspeed via its `deepspeed.initialize` `args`. There might be other args down the road.

But I hear you - we want to indicate what is expected inside `args` at `deepspeed.initialize` - though shall be done.

Reply:
Done.